### ISSUE TITLE
[BUG] [Immer 5.0.x] Uncaught TypeError: undefined is not a function (Object.assign)

### ISSUE THREAD

**Issue by FKobus** [immerjs/immer#482]
## 🐛 Bug Report

On older browsers (mobile) the immer polyfill for Object.assign doesn't get polyfilled correctly.
The error thrown is super generic. But with debugging boils down to this code:

```
function makeTrapsForGetters(getters) {
  return Object.assign({}, reflectTraps, {
    get: function get(state, prop, receiver) {
      return getters.hasOwnProperty(prop) ? getters[prop](state, prop, receiver) : Reflect.get(state, prop, receiver);
    },
    setPrototypeOf: function setPrototypeOf(state) {
      throw new Error("Object.setPrototypeOf() cannot be used on an Immer draft"); // prettier-ignore
    }
  });
}
```

## Link to repro

Our repro is private.

## To Reproduce

Just run immer on an old browser.

We used a HTC ONE M9
Chrome version:  40.0.2214.89

## Observed behavior

Crashes completely the application

## Expected behavior

Custom polyfill works.

## Environment

- **Immer version:**
- [x] Occurs with `setUseProxies(true)`
- [x] Occurs with `setUseProxies(false)` (ES5 only)

## Workaround,
Use immer 4.0.2. There the polyfill functions as expected.

This issue is created because of this issue that was closed: https://github.com/immerjs/immer/issues/477#issuecomment-565984688

**Comment by mweststrate** [comment]
This code isn't in the sources, so must relate to the transpiler setup :/

**Comment by mweststrate** [comment]
@egoist ideas?

**Comment by egoist** [comment]
@mweststrate You're using object rest spread ing here so it gets transpiled to `Object.assign`:

https://github.com/immerjs/immer/blob/7a8e4e4cfdb5a47a3dd76ca672c77e69fc417a98/src/proxy.js#L364

I guess you should change this to use your `assign` utility:

https://github.com/immerjs/immer/blob/7a8e4e4cfdb5a47a3dd76ca672c77e69fc417a98/src/common.js#L64

**Comment by mweststrate** [comment]
@egoist yeah that is what I assumed, but I guess that means that the transpilation config is off, as that should prevent us against such mistakes? Probably the target ES version is too high? If transpilation doesn't guard us against this we will run into it every time a spread is introduced :) IIRC babel shout emit utility functions for this automatically?

**Comment by mweststrate** [comment]
Pushed a patch, will revisit when finalizing TS conversion. @FBKobus could you confirm version 5.0.02 fixes the issue?

**Comment by FKobus** [comment]
@mweststrate I will test tomorrow on several old devices and report here the outcome.
Looking at the MR it seems to resolve to your internal polyfill, so i'm suspecting green flags

**Comment by FKobus** [comment]
@mweststrate ; hereby I confirm that the bug has been resolved in version 5.0.2.

**Comment by mweststrate** [comment]
Thanks!
